hpx/util/detail/pack_traversal_async_impl.hpp

```diff
@@ -436,9 +457,14 @@ namespace util {
             template <typename Begin, typename Sentinel>
             void async_traverse(dynamic_async_range<Begin, Sentinel> range)
             {
-                for (; !range.is_finished() && !is_detached(); ++range)
+                if (!is_detached())
                 {
-                    async_traverse_one(range);
+                    for (/**/; !range.is_finished(); ++range)
+                    {
+                        async_traverse_one(range);
+                        if (is_detached())          // test before increment
```

Comment:
What I'm wondering here about, is that the iterator isn't modified inside `async_traverse_one`.
When `!range.is_finished()` it should be perfectly valid to increment the iterator.
In the worst case we finish the range but in this case it should never yield an iterator dereferencation error.

Reply:
Right, the iterators are not changed inside `async_traverse_one`. However this function might move away the vector those iterators refer to (see the `std::move(args_)` here: https://github.com/STEllAR-GROUP/hpx/blob/f1285f3b46b6ca5216906350329d0a75648d8704/hpx/util/detail/pack_traversal_async_impl.hpp#L176). This was actually the main problem I stumbled over.